Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM python:3.12.3-slim
+LABEL maintainer="94nj111@gmail.com"
+
+ENV PYTHONDONTWRITEBYTECODE=True
+ENV PYTHONUNBUFFERED=True
+
+WORKDIR /app
+
+ARG UID=10001
+RUN adduser \
+    --disabled-password \
+    --gecos "" \
+    --home "/app" \
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid "${UID}" \
+    appuser
+
+RUN pip install --no-cache-dir requests
+
+USER appuser
+
+COPY app/main.py /app/main.py
```

Comment:
Ensure that the `app/main.py` file exists in your build context and is correctly copied into the Docker image. If there are additional dependencies or files, consider copying them as well.